### PR TITLE
Feature: Renaming lexer grammars (Issue #433)

### PR DESCRIPTION
**Pull request by kjarrio**
Signed-off-by: Kjartan Olason <olasonk@gmail.com>

@@ -1,5 +1,23 @@

### CODE REVIEW

**Comment by kjarrio** [comment]
Please wait before merging this one, I need to test this feature a little bit better, just a couple of edge cases. Thanks.

**Comment by parrt** [comment]
Seems like a big change, which makes me nervous as I don't have too much time to think about this project at moment ;)

**Comment by kjarrio** [comment]
Yes, although I haven't seen any issue with this while using it this week I think we should let this wait for now

**Comment by kjarrio** [comment]
btw, I have a couple of more features in the pipeline, one is larger and the other is smaller, but both are quite useful:

1. Language injection for Java - working, but still need to finish a couple of more tests. (Issue #377)
2. Context aware Completion - almost ready (#367 and #284)

I will make PR for those soon, but both are big changes.

**Comment by parrt** [comment]
OK well let's be careful about what we had; also remember I don't have a huge amount of time and the plug-in so big changes will likely sit for a long time.

**Comment by kjarrio** [comment]
Sure! Most of these changes I use for editing my own grammars, but will make those changes available as a PR though, for the community. No worries whether they get merged or not, now or later, they'll be here if needed.
